saleor/graphql/tax/mutations/tax_country_configuration_update.py

```diff
@@ -121,44 +126,93 @@ def _clean_tax_class_ids(tax_rate_items):
+                    country=country_code,
+                    tax_class=None,
+                    defaults={"rate": default_rate["rate"]},
+                )
+            else:
+                default_rate_obj = models.TaxClassCountryRate.objects.filter(
+                    country=country_code,
+                    tax_class=None,
+                ).first()
+                if default_rate_obj:
+                    default_rate_obj.delete()
 
-        # create new instances
-        to_create = [
-            models.TaxClassCountryRate(
-                country=country_code, tax_class_id=tax_class_id, rate=item["rate"]
-            )
-            for tax_class_id, item in cleaned_data.items()
-            if tax_class_id not in update_tax_classes and tax_class_id is not None
-        ]
+    @classmethod
+    def update_and_create_country_rates(cls, country_code, cleaned_data):
+        # Prepare IDs to create and update.
+        input_ids = [key for key in cleaned_data.keys() if key is not None]
+        update_qs = models.TaxClassCountryRate.objects.filter(
+            country=country_code, tax_class_id__in=input_ids
+        )
+        update_ids = [item.tax_class_id for item in update_qs]
+        create_ids = set(input_ids) - set(update_ids)
+        delete_ids = []
+
+        # Update existing instances.
+        for obj in update_qs:
+            rate = cleaned_data[obj.tax_class_id].get("rate")
+            if rate is None:
+                delete_ids.append(obj.tax_class_id)
+            else:
+                obj.rate = rate
+        models.TaxClassCountryRate.objects.bulk_update(update_qs, fields=("rate",))
+
+        # Create new instances.
+        to_create = []
+        for tax_class_id in create_ids:
+            input_item = cleaned_data[tax_class_id]
+            rate = input_item.get("rate")
+            if rate is None:
+                code = (
+                    TaxCountryConfigurationUpdateErrorCode.CANNOT_CREATE_WITH_NULL_RATE
+                )
+                raise ValidationError(
+                    {
+                        "update_tax_class_rates": ValidationError(
+                            code=code,
+                            message=(
+                                "Cannot create a new tax rate without providing it's "
+                                "value."
+                            ),
+                            params={
+                                "tax_class_ids": [
+                                    graphene.Node.to_global_id(
+                                        "TaxClass", tax_class_id
+                                    ),
+                                ]
+                            },
+                        )
+                    }
+                )
+            else:
+                obj = models.TaxClassCountryRate(
+                    country=country_code, tax_class_id=tax_class_id, rate=rate
+                )
+                to_create.append(obj)
```

Comment:
IMO we can create two methods from that: one for the creation and the second for the update.